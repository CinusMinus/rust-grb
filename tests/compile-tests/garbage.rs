use gurobi::*;
mod utils;

fn main() -> Result<()> {
  let _g = gag::Gag::stdout();
  create_model!(_g, m, x, y, z);

  c!(x + y + 2*z + 1 <= 0 in 1..0);
  c!(z + y in 0..z);
  c!(y in x..z);

  Ok(())
}
