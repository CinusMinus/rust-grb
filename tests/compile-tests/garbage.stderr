error[E0277]: the trait bound `Expr: From<bool>` is not satisfied
 --> $DIR/garbage.rs:8:6
  |
8 |   c!(x + y + 2*z + 1 <= 0 in 1..0);
  |      ^ the trait `From<bool>` is not implemented for `Expr`
  |
  = help: the following implementations were found:
            <Expr as From<&T>>
            <Expr as From<LinExpr>>
            <Expr as From<QuadExpr>>
            <Expr as From<Var>>
          and 12 others
  = note: required by `from`

error[E0605]: non-primitive cast: `Var` as `f64`
 --> $DIR/garbage.rs:9:18
  |
9 |   c!(z + y in 0..z);
  |                  ^ an `as` expression can only be used to convert between primitive types or to coerce to a specific trait object

error[E0605]: non-primitive cast: `Var` as `f64`
  --> $DIR/garbage.rs:11:14
   |
11 |   c!(y in x..z);
   |              ^ an `as` expression can only be used to convert between primitive types or to coerce to a specific trait object

error[E0605]: non-primitive cast: `Var` as `f64`
  --> $DIR/garbage.rs:11:11
   |
11 |   c!(y in x..z);
   |           ^ an `as` expression can only be used to convert between primitive types or to coerce to a specific trait object
